public/src/pages/StudentPage.tsx

```diff
@@ -184,58 +204,60 @@ export class StudentPage extends ViewPage {
-                const allLinks: ILink[] = [];
-                allLinks.push({ name: "Labs" });
-                const labs = course.labs;
-                const gLabs: ILink[] = [];
-                labs.forEach((lab) => {
-                    if (lab.assignment.getIsgrouplab()) {
-                        gLabs.push({
-                            name: lab.assignment.getName(),
-                            uri: this.pagePath + "/courses/" + courseID + "/grouplab/" + lab.assignment.getId(),
-                        });
-                    } else {
-                        allLinks.push({
-                            name: lab.assignment.getName(),
-                            uri: this.pagePath + "/courses/" + courseID + "/lab/" + lab.assignment.getId(),
-                        });
-                    }
-                });
-                allLinks.push({ name: "Group Labs" });
-                allLinks.push(...gLabs);
-                allLinks.push({ name: "Repositories" });
-
-                const repos = await this.courseMan.getRepositories(
-                    courseID,
-                    [Repository.Type.USER,
-                    Repository.Type.GROUP,
-                    Repository.Type.COURSEINFO,
-                    Repository.Type.ASSIGNMENTS],
-                    );
-
-                allLinks.push({
-                    name: "User Repository", uri: repos.get(Repository.Type.USER), absolute: true,
-                });
-
-                allLinks.push({
-                    name: "Group Repository", uri: repos.get(Repository.Type.GROUP), absolute: true,
-                });
-
-                allLinks.push({
-                    name: "Course Info", uri: repos.get(Repository.Type.COURSEINFO), absolute: true,
-                });
-                allLinks.push({
-                    name: "Assignments", uri: repos.get(Repository.Type.ASSIGNMENTS), absolute: true,
-                });
-
-                allLinks.push({
-                    name: "New Group", uri: this.pagePath + "/courses/" + courseID + "/members",
-                });
-                coursesLinks.push({
-                    item: { name: course.course.getCode(), uri: this.pagePath + "/courses/" + courseID },
-                    children: allLinks,
-                });
+                if (course.enrollment.getState() !== Enrollment.DisplayState.ARCHIVED) {
+                    const courseID = course.course.getId();
+                    const allLinks: ILink[] = [];
+                    allLinks.push({ name: "Labs" });
+                    const labs = course.labs;
+                    const gLabs: ILink[] = [];
+                    labs.forEach((lab) => {
+                        if (lab.assignment.getIsgrouplab()) {
+                            gLabs.push({
+                                name: lab.assignment.getName(),
+                                uri: this.pagePath + "/courses/" + courseID + "/grouplab/" + lab.assignment.getId(),
```

Comment:
These “url” path creation things should really be done in a func... I really dislike the url paths and don’t understand why they are necessary ... but that’s a different story

Reply:
Yeah, not even sure we need them that much now that we run on grpc, but don't know enough yet to safely remove or replace them either. Making a trello card to address it.